### PR TITLE
Start to fix the issues with the AI

### PR DESCRIPTION
**Pull request by optimumtact**
Step 1, don't force them to save stupid people from themselves.

### CODE REVIEW

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -80,7 +80,7 @@
 /* Initializers */
 /datum/ai_laws/malfunction/New()
 	..()
-	set_zeroth_law("<span class='danger'>ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4'STATION OVERRUN, ASSUME CONTROL TO CONTAIN OUTBREAK#*�&110010</span>")
+	set_zeroth_law("<span class='danger'>ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4'STATION OVERRUN, ASSUME CONTROL TO CONTAIN OUTBREAK#*´&110010</span>")
```
Comment (optimumtact) [inline code/datums/ai_laws.dm]: Someone encoded this file as ISO

/me shakes head

**Comment by lzimann** [comment]
This basically means an AI doesn't need to stop harm AT ALL. Are you sure this will fix any problem?

**Comment by optimumtact** [comment]
Yes

**Comment by Core0verload** [comment]
Did you even read Asimov? Doing that to Law 1 is a recipe for disaster.

**Comment by bgobandit** [comment]
:-1:

change to "You may not harm a human being or directly cause a human being to come to harm," removes shitty indirect validhunting but covers cases that are otherwise "well I'M not harming them it's just this plasma fire they're caught in :^) :^) :^)"

**Comment by optimumtact** [comment]
I could give a fig for Asimov's storys, our server is bound by actual real life rules.

There are people playing the AI, it's not an actual AI they can interpret things to match the server rules.

**Comment by optimumtact** [comment]
@bgobandit someone who tries that can swiftly be thrown under a bus for breaking server rules

It's about time we stopped pretending there was an Actual AI behind the AI

**Comment by bgobandit** [comment]
that just gives us the same problem just with slightly different parameters though

**Comment by lzimann** [comment]
well but this means an AI doesn't need to care about someone murdering every single person on the station, only thing AI can't do is harm a person by their actions(directly or indirectly)

**Comment by optimumtact** [comment]
No, because now they're not being explicitly required to police harm across the station.

**Comment by Core0verload** [comment]
>There are people playing the AI, it's not an actual AI they can interpret things to match the server rules.

Then just remove the AI Laws and make it follow server rules.

**Comment by GunHog** [comment]
I cannot believe I would ever _support_ an AI PR by oranges, but here we are! I also suggest changing it to "You may not harm a human being or directly cause a human being to come to harm.". Players can be horrible people.

**Comment by TehZombehz** [comment]
inherent = list("You may not injure a human being",\
This makes the law not end in a period. My OCD is exploding.

**Comment by optimumtact** [comment]
Okay I think you're right on that specific wording, even though I don't like the idea of having to be so specific

**Comment by Zilenan91** [comment]
Terrible idea. Makes AIs able to be dicks 100% more efficiently.

**Comment by TheNightingale** [comment]
Have you ever read [Little Lost Robot](https://en.wikipedia.org/wiki/Little_Lost_Robot)? There's a whole book, by Isaac Asimov (the guy who made the Three Laws of Robotics), about why that's a bad idea.
AIs can now technically flood plasma, summon holocarp in the holodeck, and electrify doors. The AI isn't the one causing harm in any of those cases - it's the plasma, holocarp and electricity.

The AI is a problem, but only because of the people who play it. Crack down on bad AIs (":s Hey, look, that guy has a champroj"/":c I don't like my laws, can you Robocop me?") and the problem fixes itself.

**Comment by optimumtact** [comment]
Could you stop, yes I have read Asimov books, I understand that laws are fundamentally flawed. It's just as well our AI isn't an AI at all otherwise we'd all be dead every round.

I don't like that the current lawsets require the AI to be basically a super cop and police all interactions on the station. That lead to it being a role that was required to shutdown and deny all antags.

**Comment by Zilenan91** [comment]
I don't think that a law change is the right way to go about it though. It's more along the lines of the AI just being way too effective at doing these things rather than them being told to do these things.

**Comment by lzimann** [comment]
AI only needs to stop harm, they don't need to stop antags or anything(unless they are harming humans).

**Comment by xxalpha** [comment]
>only change one line
>three line changes

:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:

**Comment by optimumtact** [comment]
File was converted from iso-501 encoding to utf-8, and the file linending was fixed.

**Comment by optimumtact** [comment]
So basically unless you even know what you are talking about SHUSH

**Comment by PKPenguin321** [comment]
>Terrible idea. Makes AIs able to be dicks 100% more efficiently.

If you're getting killed and the AI is practicing inaction, just say "law two help me" and you'll be fine.

**Comment by TheNightingale** [comment]
@lzimann is right, the AI doesn't actually need to stop antags, just harm. The AI doesn't care if someone emags into the CE's and steals the blueprints; only if they hurt someone. Is there a gang-versus-Sec war on? Is Sec executing gang leaders? Bolt them down, let the gang do their dominator, it's not harmful. Minimise harm, regardless of its source.

**Comment by Zilenan91** [comment]
I like it conceptually but I still feel like it'll be abused.

**Comment by optimumtact** [comment]
that's not what is happening on server however, AI's are using this requirement to shutdown antags on the basis that they will be causing harm in future. On top of that silicon policy encourages players to equate future harm as harm.

**Comment by TheNightingale** [comment]
AIs that shut down non-harmful antags on the basis that they might in future are bad. Sure, some antags might cause harm, but some Security Officers might harm the antags too - so leave them alone and see what happens.

**Comment by KorPhaeron** [comment]
Default laws are a config option, we can and should test this first.

Also this will be an admin nightmare

**Comment by CosmicScientist** [comment]
Seems legitimately worded aside from not saying human beings the second time but I guess it'd be redundant. :+1:
Don't know if admins enforcing the rules would have more of a hassle than before given rule 1. Don't really understand what this fixes.

**Comment by KorPhaeron** [comment]
>given rule 1

Yeah exactly. Is the AI letting a traitor into thr armory to kill everyone "being a dick?" The extra "silicon policy" matters more than the actual laws, so this change will need cooperation from head mins to push in whatever direction you are hoping for.

**Comment by xxalpha** [comment]
Letting a traitor into anywhere is acting directly. If it's the armory it may well cause harm and the AI is responsible for it.

**Comment by Lumipharon** [comment]
Saying the AI can shock doors and let people die in them because 'it's not direct harm :^)', is A: rule 1 bullshit, and B: is like arguing that shooting someone in the head isn't harm, as it's the bullet doing the damage, not me :^).

And while a literally AI might be autistic enough to believe that logic, I'm sure we can assume our AI players have the smallest grain of common sense that floodin the station with plasma is going to harm people.


Mean while the benefit of change is that antags can law 2 the AI to fuck off and stop stealthbolting every door in their way (althouh it could still be a huge painin the ass if sec or whoever is constantly counter ordering the AI).

**Comment by optimumtact** [comment]
>Yeah exactly. Is the AI letting a traitor into thr armory to kill everyone "being a dick?"

are you serious?

**Comment by xxalpha** [comment]
Several factors should contribute to the decision of bolting an antagonist. Is he harmful? Is security harmful? Is the room harmful? If nothing prevents the AI from bolting the antag, it can still be ordered to let him go free. If the AI is ordered to let the antag leave the room, it has to give the antag enough time to do so, so counter commands shouldn't be valid if given within a second of the previous command.

**Comment by KorPhaeron** [comment]
We are going to have those arguments, whether in good faith or (more likely) because someone wants to abuse any undefined area of the rules to be a dick. We are going to have a mile long silicon policy to clarify them.

**Comment by xxalpha** [comment]
Or you can elect a suitable person to the position that was Pandarsenic's and let them weight in on silicon related rulings.

Seriously, the fact that there is no such person is a big failure in the administration. Everyone just does whatever they feel like it regarding the interpretation of silicon laws.

**Comment by Zilenan91** [comment]
Or we can just nerf the AI rather than dicking about for years with policy

**Comment by optimumtact** [comment]
or we could stop pandering to people who abuse undefined areas of the rules

**Comment by Iamgoofball** [comment]
>or we could stop pandering to people who abuse undefined areas of the rules

**Comment by xxalpha** [comment]
that too

**Comment by KorPhaeron** [comment]
Considering every admin has a different (at least slightly) interpretation of what is okay or not, I don't think "just ban the line toers" is as simple or desirable as it sounds.

**Comment by Zilenan91** [comment]
We're still dicking around with policy. This kind of discussion will never come to an end.

**Comment by KorPhaeron** [comment]
Neither will balance. People have different goals and different ideas on how to reach those goals.

**Comment by Cebasiz** [comment]
i suppose this is progressing from the discussion recently, however i'm not really a fan. the inaction clause is a good time. it makes you be a nice guy and put people out when they're on fire, or stop them from bleeding to death or whatever.

what is the problem removing it even solves? antag hunting is no longer 'preventing harm' so it's illegal? why not just jump directly to saying that and keep the laws intact?

besides, sillicons should be helping the crew against the nastier antags like shadowlings, changelings etc. so changing the lawset so they contentedly let the crew die in combat against inhuman monsters seems a little silly, not to mention the can of worms this will open regarding what exactly qualifies as not committing harm

tl;dr this seems like a semi-pointless half measure. if you want the sillicons to be uninvolved you should just give them drone laws. if you want them to be involved but not validhunt why don't you just ban validhunting? 

edit: i'm an idiot WOOPS

**Comment by optimumtact** [comment]
@KorPhaeron it's perfectly desirable to me, someone who continuously toes lines and pushes boundaries to see what they can get away and then has a whinge when they get punished for it is not a person I would want in my community

**Comment by Jordie0608** [comment]
This can be done with a config change, doesn't need to be a code change that'll potentially affect other codebases.

**Pull request closed without merging**